compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -83,7 +84,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     if let Some(param_local_id) = param.def_id.as_local() {
                         let param_hir_id = tcx.hir().local_def_id_to_hir_id(param_local_id);
                         let param_name = tcx.hir().ty_param_name(param_hir_id);
-                        let param_type = tcx.type_of(param.def_id);
+                        let param_type = tcx.infer_ctxt().enter(|infcx| {
+                            infcx.resolve_numeric_literals_with_default(
+                                infcx.resolve_vars_if_possible(tcx.type_of(param.def_id)),
+                            )
+                        });
```

Comment:
`type_of` really shouldn't contain inference variables :thinking: 

for which test is this necessary?

Reply:
Apparently it wasn't 🤷